backend/code_coverage_backend/gcp.py

```diff
@@ -85,108 +75,115 @@ def __init__(self, reports_dir=None):
 
         # Load most recent reports in cache
         for repo in REPOSITORIES:
-            for rev, _ in self.list_reports(repo, nb=1):
-                self.download_report(repo, rev)
+            for report in self.list_reports(repo, nb=1):
+                self.download_report(report)
 
-    def ingest_pushes(self, repository, min_push_id=None, nb_pages=3):
+    def ingest_pushes(self, repository, platform, suite, min_push_id=None, nb_pages=3):
         """
         Ingest HGMO changesets and pushes into our Redis Cache
         The pagination goes from oldest to newest, starting from the optional min_push_id
         """
-        chunk_size = 8
-        params = {"version": 2}
-        if min_push_id is not None:
-            assert isinstance(min_push_id, int)
-            params["startID"] = min_push_id
-            params["endID"] = min_push_id + chunk_size
-
-        for page in range(nb_pages):
-
-            r = requests.get(
-                HGMO_PUSHES_URL.format(repository=repository), params=params
-            )
-            data = r.json()
-
-            # Sort pushes to go from oldest to newest
-            pushes = sorted(
-                [(int(push_id), push) for push_id, push in data["pushes"].items()],
-                key=lambda p: p[0],
-            )
-            if not pushes:
-                return
-
-            for push_id, push in pushes:
-
-                changesets = push["changesets"]
-                date = push["date"]
-                self.store_push(repository, push_id, changesets, date)
-
-                reports = [
-                    changeset
-                    for changeset in changesets
-                    if self.ingest_report(repository, push_id, changeset, date)
-                ]
-                if reports:
-                    logger.info("Found reports in that push", push_id=push_id)
-
-            newest = pushes[-1][0]
-            params["startID"] = newest
-            params["endID"] = newest + chunk_size
-
-    def ingest_report(self, repository, push_id, changeset, date):
+        ingested = False
+        for push_id, push in hgmo_pushes(repository, min_push_id, nb_pages):
+            for changeset in push["changesets"]:
+                # TODO: look all neighboring reports on GCP
```

Comment:
?

Reply:
It was an idea before adding platform & suite support in that function, to lookup all the combinations with the repo+revision prefix on gcp and ingest them at once. But it's useless now.